### PR TITLE
Rename `RatchetKey` to `RevisionKey`, encrypt `RevisionKey` within `PrivateRef`

### LINKED ISSUE #77: Fix temporal encryption scheme
In the definition for `PrivateRef`:
https://github.com/wnfs-wg/rs-wnfs/blob/11b73a7be75f243b09e0147dec8171cc06e8c786/wnfs/private/node.rs#L93-L102

Serialization is derived via the macro.

This struct is used in the `PrivateDirectorySerde` representation of directories:
https://github.com/wnfs-wg/rs-wnfs/blob/11b73a7be75f243b09e0147dec8171cc06e8c786/wnfs/private/directory.rs#L51-L58

Again, serialization is derived using the macros.

However, we actually want the `ratchet_key` within the `entires` of directories to be encrypted with the directory's `ratchet_key`!

Again, we have an issue with passing around `key`s to enable serialization.
We could define & use a `PrivateRefSerde` in `PrivateDirectorySerde` and decrypt it on-the-fly similarly to the `PrivateDirectorySerde`.

### PR DESCRIPTION
**Pull request by matheus23**
Closes #77 

This also renames `RatchetKey` to `RevisionKey`, to be more in line with the terminology from the spec.

The `RevisionKey` within `PrivateRef`s in directories is meant to be encrypted with the directory's `RevisionKey`.

### CODE REVIEW

File: wnfs/private/forest.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ impl PrivateForest {
     /// # Examples
     ///
     /// ```
-    /// use std::rc::Rc;
+    /// use std::{collections::BTreeSet, rc::Rc};
```
[the remark this answers is not in the document]
Comment (matheus23) [reply wnfs/private/forest.rs:51]: Wait. No! Thumbs down! :-1: 

Sorry there's some diff that made it in here. Fixing this.

[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply wnfs/private/forest.rs]: it was oh really thumbs haha, cause seemed out of place haha.

**Comment by matheus23** [comment]
Messed up the branching & merging. Please stand by while I fix this PR

**Comment by matheus23** [comment]
Flipped this to 'ready for review'. No more merge conflicts & no more rogue changes.

File: wnfs/src/private/node.rs [hunk 2]
```diff
@@ -612,18 +622,82 @@ impl From<ContentKey> for Key {
...
         Self {
             saturated_name_hash,
-            content_key: ratchet_key.derive_content_key(),
-            ratchet_key,
+            content_key: revision_key.derive_content_key(),
+            revision_key,
         }
     }
+
+    pub(crate) fn to_serde(
```
Comment (appcypher) [inline wnfs/src/private/node.rs:633]: `to_serde` (`from_serde`) threw me off a bit here cos there is nothing `serde` about them. 
Maybe we need to rename these structs but I can't think of a good name yet. Perhaps `to_serializable` and `PrivateRefSerializable`.

Comment (matheus23) [reply wnfs/src/private/node.rs:633]: Yeah you're right.
I also briefly thought about something involving `flush`, but that's more true for the public stuff than the private stuff. :thinking: 

I like the `*Serializable` suffix better than the `Serde` suffix.

Comment (matheus23) [reply wnfs/src/private/node.rs]: I couldn't come up with anything better, so for now I've renamed all `*Serde` structs into `*Serializable` structs.

Comment (matheus23) [reply wnfs/src/private/node.rs]: I also renamed the `to_serde` and `from_serde` functions into `*_serializable`.

File: wnfs/src/private/directory.rs [hunk 3]
```diff
@@ -1150,9 +1150,18 @@ impl PrivateDirectory {
...
             .get_private_ref()
             .map_err(SerError::custom)?
-            .ratchet_key;
+            .revision_key;
+
+        let mut entries = BTreeMap::new();
 
-        (PrivateDirectorySerde {
+        for (name, private_ref) in self.entries.iter() {
+            let private_ref_serde = private_ref
```
Comment (zeeshanlakhani) [inline wnfs/src/private/directory.rs:1158]: super minor: for consistency:

```suggestion
let private_ref_serializable = private_ref
```

**Review by zeeshanlakhani — APPROVED** [review]
super minor naming thing that matters very little, 👍🏽

**Review by appcypher — APPROVED** [review]
<div align="center">
    <a href="#" target="_blank">
        </img>
    </a>
</div>

<h3 align="center">LGTM! 🎉 </h3>

**Pull request merged**